Subgradients.thy

```diff
@@ -0,0 +1,371 @@
+theory Subgradients
```

Comment:
please adhere to [theory documentation guidelines](https://isabelle.systems/conventions/theory_documentation.html), at least add creator, summary and main definitions/theorems